fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java

```diff
@@ -1043,20 +1044,29 @@ private void getClusterIdAndRole() throws IOException {
                     System.exit(-1);
                 }
             }
-
             getNewImage(rightHelperNode);
         }
 
+        if (isFirstTimeStartUp) {
+            Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+                String distDir = this.imageDir + "." + System.currentTimeMillis();
+                LOG.info("first time start failed, move image dir to {}", distDir);
+                if (listener == null) {
```

Comment:
Using listener to judge the start failure is too tricky!!!

Reply:
Moving meta data is a very dangerous operation. We should thick twice to fix the unusual problem.

Reply:
How about 

> Moving meta data is a very dangerous operation. We should thick twice to fix the unusual problem.

Yes you're right.
I want to check port using before create meta files, but port may occupied later, and process may exit caused by other error.
Do you have any suggestions?

Reply:
> Check if edit port is used before getClusterIdAndRole？

There is a minor gap between check port using and init bdb, is it acceptable ?

Reply:
> There is a minor gap between check port using and init bdb

I think it can solve most scenarios.